modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,248 @@
 /*
- * PROJECT:         ReactOS api tests
- * LICENSE:         GPL - See COPYING in the top level directory
- * PURPOSE:         Test for NtUserSetTimer
- * PROGRAMMERS:
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for NtUserSetTimer
+ * COPYRIGHT:   Copyright 2008 Timo Kreuzer <timo.kreuzer@reactos.org>
+ *              Copyright 2024 Tomáš Veselý <turican0@gmail.com>
  */
 
 #include "../win32nt.h"
 
+#define SLEEP_TIME 500
+#define TIME_TOLERANCE 0.5
+
+#define TEST1_COUNT 20
+#define TEST1_INTERVAL 10
+
+#define TEST2_COUNT 40000
+#define TEST2_INTERVAL 10
+
+#define TESTW1_COUNT 20
+#define TESTW1_INTERVAL 10
+
+#define TESTW2_COUNT 40000
+#define TESTW2_INTERVAL 10
+
+typedef struct TIMER_MESSAGE_STATE1
+{
+    UINT_PTR index;
+    UINT counter;
+} TIMER_MESSAGE_STATE1;
+
+TIMER_MESSAGE_STATE1 timerId1[TEST1_COUNT];
+
+typedef struct TIMER_MESSAGE_STATEW1
+{
+    UINT counter;
+} TIMER_MESSAGE_STATEW1;
+
+TIMER_MESSAGE_STATEW1 timerIdW1[TESTW1_COUNT];
+
+void CALLBACK TimerProc(HWND hWnd, UINT uMsg, UINT_PTR idEvent, DWORD dwTime)
+{
+    UINT i;
+    for (i = 0; i < TEST1_COUNT; i++)
+    {
+        if (timerId1[i].index == idEvent)
+            timerId1[i].counter++;
+    }
+}
+
+void MessageLoop(void)
```

Comment:
This function doesn't appear to be used anywhere. If you confirm to be so, please remove it.

Reply:
You're right, I ended up integrating that feature directly. Removed.